### PR TITLE
feature: the pipeline yml is updated need to modify pipeline cron

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
feature: the pipeline yml is updated need to modify pipeline cron

#### Which issue(s) this PR fixes:
https://erda-org.erda.cloud/erda/workBench/projects/387/issues/all?id=66691&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMTI2MSJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=431&type=BUG

https://erda-org.erda.cloud/erda/workBench/projects/387/issues/all?id=70714&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMTI2MSJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=431&type=BUG

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/event.go [hunk 1]
```diff
@@ -226,3 +226,12 @@ func (ie *IssueEvent) GenEventParams(locale, uiPublicURL string) map[string]stri
 
 	return params
 }
+
+type PayloadPushEvent struct {
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply apistructs/event.go]: done

File: modules/dop/endpoints/endpoints.go [hunk 2]
```diff
@@ -214,6 +214,7 @@ func (e *Endpoints) Routes() []httpserver.Endpoint {
 		{Path: "/api/cicd-crons/{cronID}/actions/start", Method: http.MethodPut, Handler: e.pipelineCronStart},
 		{Path: "/api/cicd-crons/{cronID}/actions/stop", Method: http.MethodPut, Handler: e.pipelineCronStop},
 		{Path: "/api/cicd-crons", Method: http.MethodPost, Handler: e.pipelineCronCreate},
+		{Path: "/api/cicd-crons/update", Method: http.MethodPost, Handler: e.pipelineCronUpdate},
```
Comment (sfwn) [inline modules/dop/endpoints/endpoints.go:217]: PUT: /api/cicd-crons/{cronID}

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/endpoints/endpoints.go (hunk 2 above)
Comment (littlejiancc) [reply modules/dop/endpoints/endpoints.go]: eventBox call back only support post method
and the path is updated to /api/cicd-crons/actions/hook-for-update

**Review by sfwn — APPROVED** [review]

**Pull request merged**